src/benchmarks/micro/runtime/JetStream/Poker/poker.cs

```diff
@@ -0,0 +1,366 @@
+                    this._rank = (this._rank << 4) | cardRank;
+                }
+            }
+        }
+
+        public int rank {
+            get {
+                return _rank;
+            }
+        }
+
+        public override string ToString () {
+            return this._cards.join("");
+        }
+    }
+
+    public class Player : Hand {
+        public string _name;
+        public int _wins;
+        public int[] _handTypeCounts;
+
+        public Player (string name)
+            : base () {
+            this._name = name;
+            this._wins = 0;
+            this._handTypeCounts = new int[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
+        }
+
+        public void scoreHand() {
+            this.score();
+            var handType = this.rank >> 20;
+            this._handTypeCounts[handType]++;
+        }
+
+        public void wonHand() {
+            this._wins++;
+        }
+
+        public string name {
+            get {
+                return this._name;
+            }
+        }
+
+        public string hand {
+            get {
+                return base.ToString();
+            }
+        }
+
+        public int wins {
+            get {
+                return this._wins;
+            }
+        }
+
+        public int[] handTypeCounts {
+            get {
+                return this._handTypeCounts;
+            }
+        }
+    }
+
+    public static class Hands {
+        public static void playHands (List<Player> players)
+        {
+            const int handCount = 2000;
+
+            var cardDeck = new CardDeck();
+            var handsPlayed = 0;
+            var highestRank = 0;
+
+            do {
+                cardDeck.shuffle();
+
+                foreach (var player in players)
+                    player.clear();
+
+                for (var i = 0; i < 5; i++) {
+                    foreach (var player in players)
+                        player.takeCard(cardDeck.dealOneCard());
+                }
+
+                foreach (var player in players)
+                    player.scoreHand();
+
+                handsPlayed++;
+
+                highestRank = 0;
+
+                foreach (var player in players)
+                    if (player.rank > highestRank)
+                        highestRank = player.rank;
+
+                foreach (var player in players)
+                    // We count ties as wins for each player.
+                    if (player.rank == highestRank)
+                        player.wonHand();
+
+            } while (handsPlayed < handCount);
+        }
+    }
+
+    [BenchmarkCategory(Categories.Runtime, Categories.JetStream)]
+    public class PokerBenchmark {
+        public List<Player> _players = new List<Player> {
+            new Player("Player 1"),
+            new Player("Player 2"),
+            new Player("Player 3"),
+            new Player("Player 4")
+        };
+
+        [Benchmark(Description = nameof(PokerBenchmark))]
+        public void runIteration()
```

Comment:
the benchmark ID that is used to store the data in the reporting system and also for filtering consists of: `namespace.TypeName.MethodName`. We try to [avoid renaming benchmarks as much as possible](https://github.com/dotnet/performance/blob/master/docs/microbenchmark-design-guidelines.md#benchmarks-are-immutable), but since you are adding a new benchmark and the ID does not exist in our reporting system then I think that the names could be simplified now:

```diff
- JetStream.PokerBenchmark.runIteration
+ JetStream.Poker.Play
```

```suggestion
    [BenchmarkCategory(Categories.Runtime)]
    public class Poker {
        public List<Player> _players = new List<Player> {
            new Player("Player 1"),
            new Player("Player 2"),
            new Player("Player 3"),
            new Player("Player 4")
        };

        [Benchmark]
        public void Play()
```